tf2_ros/src/static_transform_component.cpp

```diff
@@ -0,0 +1,89 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "tf2_ros/static_transform_component.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+using namespace std::chrono_literals;
+
+namespace tf2_ros
+{
+StaticTransformPublisher::StaticTransformPublisher(
+  rclcpp::NodeOptions options, const char * name
+)
+: rclcpp::Node(name, options)
+{
+  rclcpp::ParameterValue default_value(static_cast<double>(0));
+  rcl_interfaces::msg::ParameterDescriptor descriptor;
+  descriptor.read_only = true;
+  auto tx = this->declare_parameter("/translation/x", default_value, descriptor);
+  auto ty = this->declare_parameter("/translation/y", default_value, descriptor);
+  auto tz = this->declare_parameter("/translation/z", default_value, descriptor);
+  auto rx = this->declare_parameter("/rotation/x", default_value, descriptor);
+  auto ry = this->declare_parameter("/rotation/y", default_value, descriptor);
+  auto rz = this->declare_parameter("/rotation/z", default_value, descriptor);
+  auto rw = this->declare_parameter("/rotation/w", default_value, descriptor);
+  auto p_frame_id =
+    this->declare_parameter("/frame_id", rclcpp::ParameterValue(std::string("/frame")), descriptor);
+  auto p_child_id =
+    this->declare_parameter("/child_frame_id", rclcpp::ParameterValue(std::string(
+        "/child")), descriptor);
+
+  m_broadcaster = std::make_unique<tf2_ros::StaticTransformBroadcaster>(this);
+  auto clock = std::make_shared<rclcpp::Clock>(RCL_ROS_TIME);
+  rclcpp::TimeSource time_source;
+  time_source.attachClock(clock);
+
+  geometry_msgs::msg::TransformStamped tf_msg;
+  tf_msg.transform.translation.x = tx.get<double>();
+  tf_msg.transform.translation.y = ty.get<double>();
+  tf_msg.transform.translation.z = tz.get<double>();
+  tf_msg.transform.rotation.x = rx.get<double>();
+  tf_msg.transform.rotation.y = ry.get<double>();
+  tf_msg.transform.rotation.z = rz.get<double>();
+  tf_msg.transform.rotation.w = rw.get<double>();
+  tf_msg.header.frame_id = p_frame_id.get<std::string>();
+  tf_msg.child_frame_id = p_child_id.get<std::string>();
+  // check frame_id != child_frame_id
+  if (tf_msg.header.frame_id == tf_msg.child_frame_id) {
+    RCLCPP_ERROR(this->get_logger(),
+      "cannot publish static transform from '%s' to '%s', exiting",
+      tf_msg.header.frame_id.c_str(), tf_msg.child_frame_id.c_str());
+    rclcpp::shutdown();
```

Comment:
Instead of `rclcpp::shutdown` and a return here, I think an exception is more in order.  At least that will show up in the component container.